src/lib/credits.ts

```diff
@@ -0,0 +1,625 @@
+      maxUncreditedValue,
+      maxCreditedValue
+    }
+    // Send the transaction
+    return sendTransaction(contract, 'useCredits', useCreditsArgs)
+  }
+
+  prepareCreditsCollectionStore(
+    item: Item,
+    walletAddress: string,
+    credits: Credit[]
+  ): {
+    contract: ContractData
+    creditsData: CreditsData[]
+    creditsSignatures: string[]
+    externalCall: ExternalCallParams
+    maxUncreditedValue: string
+    maxCreditedValue: string
+  } {
+    // Prepare common credits data
+    const {
+      contract,
+      creditsData,
+      creditsSignatures
+    } = this.prepareCreditsData(credits, item.chainId)
+
+    // Get the CollectionStore contract address
+    const collectionStoreContract = getContract(
+      ContractName.CollectionStore,
+      item.chainId
+    )
+    const collectionStoreAddress = collectionStoreContract.address
+
+    // Create a contract interface for the CollectionStore to get the function selector
+    const collectionStoreInterface = new ethers.utils.Interface(
+      collectionStoreContract.abi
+    )
+
+    // The selector for the buy function in the CollectionStore contract
+    const buySelector = collectionStoreInterface.getSighash('buy')
+
+    // Create the ItemToBuy structure as shown in the tests
+    const itemsToBuy = [
+      {
+        collection: item.contractAddress,
+        ids: [item.itemId],
+        prices: [item.price],
+        beneficiaries: [walletAddress]
+      }
+    ]
+
+    // Encode the buy function parameters using abi.encode
+    const buyData = ethers.utils.defaultAbiCoder.encode(
+      [
+        'tuple(address collection, uint256[] ids, uint256[] prices, address[] beneficiaries)[]'
+      ],
+      [itemsToBuy]
+    )
+
+    // Prepare the external call
+    const externalCall = this.prepareExternalCall({
+      target: collectionStoreAddress,
+      selector: buySelector,
+      data: buyData
+      // chainId: item.chainId
+    })
+
+    const creditsValue = credits.reduce(
+      (acc, credit) => acc + parseInt(credit.amount),
+      0
+    )
+    const whatUserHasToPay = BigInt(item.price) - BigInt(creditsValue)
+    const maxUncreditedValue =
+      whatUserHasToPay < BigInt(0) ? '0' : whatUserHasToPay.toString()
+
+    return {
+      contract,
+      creditsData,
+      creditsSignatures,
+      externalCall,
+      maxUncreditedValue,
+      maxCreditedValue: item.price
+    }
+  }
+
+  /**
+   * Use credits for buying items from the CollectionStore
+   * @param item - The item to buy
+   * @param walletAddress - The user's wallet address
+   * @param credits - The user's credits
+   * @returns The transaction hash
+   */
+  async useCreditsCollectionStore(
+    item: Item,
+    walletAddress: string,
+    credits: Credit[]
+  ): Promise<string> {
+    // Prepare common credits data
+    const {
+      contract,
+      creditsData,
+      creditsSignatures
+    } = this.prepareCreditsData(credits, item.chainId)
```

Comment:
The `prepareCreditsCollectionStore` and the `useCreditsCollectionStore` have a lot of common code, would you mind re-using `prepareCreditsCollectionStore` for the `useCreditsCollectionStore` function?